runtime/Python2/src/antlr4/atn/ATNConfig.py

```diff
@@ -138,11 +138,8 @@ def __eq__(self, other):
             return True
         elif not isinstance(other, LexerATNConfig):
             return False
-        if self.passedThroughNonGreedyDecision != other.passedThroughNonGreedyDecision:
-            return False
-        if self.lexerActionExecutor is not other.lexerActionExecutor:
-            return False
-        return super(LexerATNConfig, self).__eq__(other)
+        else:
+            return hash(other) == hash(self)
```

Comment:
:exclamation: This isn't how hash codes work, at least in languages like Java. In Java, if the hash codes of two objects are not equal, then you know the objects are not equal. However, if the hash codes are equal, then the objects might still not be equal. With this change, the `__eq__` method might return true in cases where the objects are not equal, leading to incorrect prediction results.